internal/filter/types.go

```diff
@@ -1,188 +1,236 @@
+// check the available exported methods.
+type Condition struct {
+	op     CompOperator
+	column string
+	value  string
 }
 
-type UnEqual Condition
-
-func (u *UnEqual) Eval(filterable Filterable) (bool, error) {
-	match, err := filterable.EvalEqual(u.column, u.value)
+// NewCondition initiates a new Condition instance from the given data.
+// Returns error if fails to url decode the provided column and/or value.
+func NewCondition(column string, operator CompOperator, value string) (Filter, error) {
+	decodedCol, err := url.QueryUnescape(column)
 	if err != nil {
-		return false, err
+		return nil, err
 	}
 
-	return filterable.EvalExists(u.column) && !match, nil
-}
-
-type Like Condition
-
-func (l *Like) Eval(filterable Filterable) (bool, error) {
-	match, err := filterable.EvalLike(l.column, l.value)
+	decodedVal, err := url.QueryUnescape(value)
 	if err != nil {
-		return false, err
+		return nil, err
 	}
 
-	return match, nil
+	return &Condition{op: operator, column: decodedCol, value: decodedVal}, nil
 }
 
-type Unlike Condition
+// Eval evaluates this Condition based on its operator.
+// Returns true when the filter evaluates to true false otherwise.
+func (c *Condition) Eval(filterable Filterable) (bool, error) {
+	var err error
+	match := false
+	switch c.op {
+	case Equal:
+		match, err = filterable.EvalEqual(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
+	case UnEqual:
+		match, err = filterable.EvalEqual(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
 
-func (u *Unlike) Eval(filterable Filterable) (bool, error) {
-	match, err := filterable.EvalLike(u.column, u.value)
-	if err != nil {
-		return false, err
-	}
+		match = filterable.EvalExists(c.column) && !match
+	case Like:
+		match, err = filterable.EvalLike(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
+	case UnLike:
+		match, err = filterable.EvalLike(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
 
-	return filterable.EvalExists(u.column) && !match, nil
-}
+		match = filterable.EvalExists(c.column) && !match
+	case LessThan:
+		match, err = filterable.EvalLess(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
+	case LessThanEqual:
+		match, err = filterable.EvalLessOrEqual(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
+	case GreaterThan:
+		match, err = filterable.EvalLessOrEqual(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
 
-type LessThan Condition
+		match = filterable.EvalExists(c.column) && !match
+	case GreaterThanEqual:
+		match, err = filterable.EvalLess(c.column, c.value)
+		if err != nil {
+			return false, err
+		}
 
-func (less *LessThan) Eval(filterable Filterable) (bool, error) {
-	match, err := filterable.EvalLess(less.column, less.value)
-	if err != nil {
-		return false, err
+		match = filterable.EvalExists(c.column) && !match
 	}
 
 	return match, nil
 }
```

Comment:
I don't understand why this is a problem now, I haven't changed anything since the previous version. I have just have reduce all the types into a single method.

Reply:
It's not like this has magically become a problem in the meantime. It's just that the old code was merged in an early phase and while in a hurry, so it was better to have some working code than fully cleaned up code. But now, since you're reworking large parts of the code, I think it makes sense to clean up these parts as well.